src/components/layout/Header.vue

```diff
@@ -27,14 +29,253 @@
       </div>
     </nav>
   </header>
+
+  <div id="mobileNavOverlay" ref="mobileNavOverlay" :class="mode">
+    <div class="blur" />
+    <ul class="mb-lg-0">
+      <li v-if="renderHomeItemInMobileNav">
+        <router-link
+          class="nav-link"
+          :to="{ name: 'Home' }"
+          :class="{
+            active: 'Home' === currentRoute,
+          }"
+          @click="toggleMobileNav"
+        >
+          {{ "Home" }}
+        </router-link>
+      </li>
+      <li v-for="view in views" :key="view">
+        <router-link
+          class="nav-link"
+          :to="{ name: view }"
+          :class="{
+            active: view === currentRoute,
+          }"
+          @click="toggleMobileNav"
+        >
+          {{ view }}
+        </router-link>
+      </li>
+    </ul>
+  </div>
 </template>
 
 <script>
-export default {
-  computed: {
-    views: function () {
-      return ['About', 'Projects', 'Blog']
-    }
+  import content from "@/ressources/components/layout/header.json"
+
+  export default {
+    props: {
+      /**
+       * As the top left h1-title in the navigation is not reachable within the mobile navigation overlay, you
+       * may want to additionally render a navigation item for the "Home" route in the mobile navigation overlay.
+       *
+       * Defaults to true.
+       */
+      renderHomeItemInMobileNav: {
+        type: Boolean,
+        default: true,
+      },
+      /**
+       * You can choose between a light mode and a dark mode.
+       * For now, the mode only applies to the mobile navigation overlay.
+       *
+       * Defaults to "light".
+       */
+      mode: {
+        type: String,
+        validator: (value) => ["light", "dark"].includes(value),
+        default: "light",
+      },
+    },
+    data: function () {
+      return {
+        isMobileNavActive: false,
+      }
+    },
+    computed: {
+      views: function () {
+        return ["About", "Projects", "Blog"]
+      },
+      currentRoute: {
+        get() {
+          return this.$route.name
+        },
+      },
+      content: function () {
+        return content
+      },
+    },
+    watch: {
+      /** Disables scrolling while mobile navigation overlay is shown. */
+      isMobileNavActive: function (currentValue) {
+        document.documentElement.style.overflow = currentValue
+          ? "hidden"
+          : "auto"
+      },
+    },
```

Comment:
Listeners bitte hier entfernen. Du könntest das overflow auch in der method `toggleMobileNav` verändern.

Reply:
Zudem frage ich mich, ob es nicht eine geschicktere Möglichkeit gibt als das Scroll-Verhalten der gesamten Seite zu verhindern? So ist deine Component nicht gekapselt und dies reduziert die Kohäsion.